pallets/parachain-staking/src/lib.rs

```diff
@@ -2252,39 +2449,40 @@ pub mod pallet {
 			candidate: T::AccountId,
 			delegator: T::AccountId,
 		) {
-			if let Ok(amount_transferred) =
-				T::Currency::deposit_into_existing(&delegator, amt.clone())
-			{
-				Self::deposit_event(Event::Rewarded {
-					account: delegator.clone(),
-					rewards: amount_transferred.peek(),
-				});
+			// Mint rewards to the delegator
+			if frame_system::Pallet::<T>::account_exists(&delegator) {
+				if let Ok(minted) = T::Currency::mint_into(&delegator, amt.clone()) {
+					Self::deposit_event(Event::Rewarded {
+						account: delegator.clone(),
+						rewards: minted,
+					});
```

Comment:
Redundant code, we can call `mint` if mint returns a Result

Reply:
Amended https://github.com/moonbeam-foundation/moonbeam/pull/3306/commits/e0d05698cbd8ba6efe95ebf9578eaca7ff7f0693